TestExamples failing: no such file

```
$ go version
go version go1.21.0 linux/amd64

$ go install github.com/ServiceWeaver/weaver/cmd/weaver@latest
go: downloading github.com/ServiceWeaver/weaver v0.19.0

$ git clone https://github.com/ServiceWeaver/weaver
$ cd weaver
$ cd examples
$ go test
--- FAIL: TestExamples (0.70s)
    --- FAIL: TestExamples/collatz (0.19s)
        --- FAIL: TestExamples/collatz/weaver-single (0.00s)
            examples_test.go:95: fork/exec ../../cmd/weaver/weaver: no such file or directory
        --- FAIL: TestExamples/collatz/weaver-multi (0.00s)
            examples_test.go:102: fork/exec ../../cmd/weaver/weaver: no such file or directory
    --- FAIL: TestExamples/factors (0.18s)
        --- FAIL: TestExamples/factors/weaver-single (0.00s)
            examples_test.go:95: fork/exec ../../cmd/weaver/weaver: no such file or directory
        --- FAIL: TestExamples/factors/weaver-multi (0.00s)
            examples_test.go:102: fork/exec ../../cmd/weaver/weaver: no such file or directory
    --- FAIL: TestExamples/hello (0.18s)
        --- FAIL: TestExamples/hello/weaver-single (0.00s)
            examples_test.go:95: fork/exec ../../cmd/weaver/weaver: no such file or directory
        --- FAIL: TestExamples/hello/weaver-multi (0.00s)
            examples_test.go:102: fork/exec ../../cmd/weaver/weaver: no such file or directory
    --- FAIL: TestExamples/reverser (0.15s)
        --- FAIL: TestExamples/reverser/weaver-single (0.00s)
            examples_test.go:95: fork/exec ../../cmd/weaver/weaver: no such file or directory
        --- FAIL: TestExamples/reverser/weaver-multi (0.00s)
            examples_test.go:102: fork/exec ../../cmd/weaver/weaver: no such file or directory
FAIL
exit status 1
FAIL    github.com/ServiceWeaver/weaver/examples        0.702s
```

The test is looking for an executable at a relative path, where it is not, and I'm not sure about a prerequisite step to copy or compile it there. My executable lives at `~/go/bin/weaver`.

--- a/examples/examples_test.go
+++ b/examples/examples_test.go
@@ -34,6 +34,13 @@ type test struct {
 }
 
 func TestExamples(t *testing.T) {
+	// Build the weaver binary.
+	cmd := exec.Command("go", "build")
+	cmd.Dir = "../cmd/weaver"
+	if err := cmd.Run(); err != nil {
+		t.Fatal(err)
+	}
+
 	testCases := map[string]test{
 		// TODO(mwhittaker): Add helloworld example.
 		"hello": {

